shared/service-registry/utils.ts

```diff
@@ -205,4 +206,55 @@ export function createDailyActiveMultisig(
     dailyActiveMultisigs.count = dailyActiveMultisigs.count + 1;
     dailyActiveMultisigs.save();
   }
+}
+
+export function createOrUpdateAgentRegistration(
+  serviceId: i32,
+  agentId: i32,
+  timestamp: BigInt
+): void {
+  const id = serviceId.toString().concat("-").concat(agentId.toString());
+  let registration = AgentRegistration.load(id);
+  if (registration == null) {
+    registration = new AgentRegistration(id);
+    registration.serviceId = serviceId;
+    registration.agentId = agentId;
+    registration.registrationTimestamp = timestamp;
+  }
+  registration.save();
+}
```

Comment:
we only update timestamps when we create the entity, is that correct? shouldn't it be something like
```suggestion
  const id = serviceId.toString().concat("-").concat(agentId.toString());
  let registration = AgentRegistration.load(id);
  if (registration == null) {
    registration = new AgentRegistration(id);
  }
  registration.serviceId = serviceId;
  registration.agentId = agentId;
  registration.registrationTimestamp = timestamp;
  registration.save();
}
```